### PR TITLE
Helm chart controller annotations

### LINKED ISSUE #30071: Helm Chart: allow setting annotations for resource controllers (CronJob, Deployment)
### Description

The Helm Chart allows setting annotations for the pods created by the `CronJob` [but not the CronJob controller itself](https://github.com/apache/airflow/blob/helm-chart/1.8.0/chart/templates/cleanup/cleanup-cronjob.yaml).

The values file should offer an option to provide custom annotations for the `CronJob` controller, similarly to how the DB migrations job exposes `.Values.migrateDatabaseJob.jobAnnotations` 

In the same fashion, other `Deployment` templates expose custom annotations, but [statsd deployment doesn't](https://github.com/apache/airflow/blob/helm-chart/1.8.0/chart/templates/statsd/statsd-deployment.yaml).

### Use case/motivation

Other tools e.g. ArgoCD may require the use of annotations, for example:
* [ArgoCD Sync Options](https://argo-cd.readthedocs.io/en/stable/user-guide/sync-options)
* [ArgoCD Sync Phases and Waves](https://argo-cd.readthedocs.io/en/stable/user-guide/sync-waves/)

Example use case:
_Set the cleanup CronJob to be synced after the webserver and scheduler deployments have been synced with ArgoCD_

### Related issues

https://github.com/apache/airflow/issues/25446 originally mentioned the issue regarding the StatsD deployment, but the accepted fix was https://github.com/apache/airflow/pull/25732 which allows setting annotations for the pod template, not the `Deployment` itself

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by bnguyennhon**
Support custom annotations for the cleanup CronJob and statsd Deployment at the controller level

Similar to how other Deployments in this chart accept `annotations` and Jobs accept `jobAnnotations`

closes: [#30071](https://github.com/apache/airflow/issues/30071)
related: [#25446](https://github.com/apache/airflow/issues/25446)

### CODE REVIEW

File: chart/templates/cleanup/cleanup-cronjob.yaml [hunk 1]
```diff
@@ -41,6 +41,9 @@ metadata:
     {{- with .Values.labels }}
       {{- toYaml . | nindent 4 }}
     {{- end }}
+  {{- if .Values.cleanup.jobAnnotations }}
+  annotations: {{- .Values.cleanup.jobAnnotations | toYaml | nindent 4 }}
+  {{- end }}
```
Comment (hussein-awala) [inline chart/templates/cleanup/cleanup-cronjob.yaml]:
```suggestion
  {{- with .Values.cleanup.jobAnnotations }}
  annotations: {{- toYaml . | nindent 4 }}
  {{- end }}
```

File: chart/templates/statsd/statsd-deployment.yaml [hunk 2]
```diff
@@ -38,6 +38,9 @@ metadata:
     {{- with .Values.labels }}
       {{- toYaml . | nindent 4 }}
     {{- end }}
+  {{- if .Values.statsd.annotations }}
+  annotations: {{- toYaml .Values.statsd.annotations | nindent 4 }}
+  {{- end }}
```
Comment (hussein-awala) [inline chart/templates/statsd/statsd-deployment.yaml]:
```suggestion
  {{- with .Values.statsd.annotations }}
  annotations: {{- toYaml . | nindent 4 }}
  {{- end }}
```

**Review by hussein-awala — CHANGES_REQUESTED** [review]
Just to make it cleaner before merging it

**Comment by bnguyennhon** [comment]
Thanks for the suggestions :)

**Review by potiuk — APPROVED** [review]

**Pull request merged**